src/js/components/DataTypes/Object.js

```diff
@@ -45,40 +47,49 @@ class rjvObject extends React.Component {
                     namespace: props.namespace
                 }) === false) &&
             //initialize closed if object has no items
-            size !== 0
+            size !== 0;
         const state = {
-            rjvId: props.rjvId,
-            state_key: props.namespace.join("."),
-            namespace: props.namespace,
-            indentWidth: props.indentWidth,
             expanded: AttributeStore.get(
                 props.rjvId,
                 props.namespace,
                 "expanded",
                 expanded
             ),
-            object_type: props.type == "array" ? "array" : "object",
-            parent_type: props.type == "array" ? "array" : "object",
-            size: size
-        }
-
-        return { ...this.state, ...state }
+            object_type: props.type === "array" ? "array" : "object",
+            parent_type: props.type === "array" ? "array" : "object",
+            size
+        };
+        return state;
```

Comment:
@mac-s-g Good question! But no, because I made sure that in this component, `expanded`, `object_type`, `parent_type`, and `size` are the only properties in the state object, and I removed all other unused/redundant state properties like `rjvId`, `state_key`, etc. Generally, we should avoid syncing props to state, because by doing that we will have two source of truth for state (one with `setState`, the other with syncing from prop).

Reply:
:+1: